easybuild/framework/easyblock.py

```diff
@@ -1857,8 +1857,15 @@ def check_checksums_for(self, ent, sub='', source_cnt=None):
             if isinstance(checksum, dict):
                 checksum = checksum.get(fn)
 
-            if not is_sha256_checksum(checksum):
-                msg = "Non-SHA256 checksum found for %s: %s" % (fn, checksum)
+            # take into account that we may encounter a tuuple of valid checksums
+            # (see https://github.com/easybuilders/easybuild-framework/pull/2958)
+            if isinstance(checksum, tuple):
```

Comment:
What about `(algo, checksum)` tuples and `{file:checksums}` dicts?

Reply:
Good point, fixed in ba413cfe3